### DISCUSSION TITLE
Updates for M1

### DISCUSSION THREAD

**Discussion by Civa69** [zurie/pysha#2] [Q&A] [answered]
I am new to this, and was searching for a way to contact Zurie, however this seems to be my only option. 

I am curious as to the state of this project, if it has been updated, does Python care if the processor is apple silicon, and if there are any plans to take the project further to give it the full functionality of Push 2 in Logic.

**Comment by zurie** [comment]
I am probably the worst person for communication! :) 3 years later.. i have never even seen this tab light up or tell me I had a message... I honestly don't use github that much... 

with that said...

the issue is not the M chip, its that OSCulator from the last time I checked was not written for M processors...  I know that some users have said that they just brought over a system that was previously on a intel mac and it worked in logic but I don't think it was official and it was almost like a fluke at best? 

I still have a i9 mac laptop that I do stuff on and haven't updated my system that my music studio runs on in a long time.   I just keep it running and have avoided updating things.. (which sucks but its the life of a audio studio alot of times).   With mac thats getting harder and harder to do because soon we will be forced to M chips...   if someone knows of OSCulator working on the M i will continue to mess with this stuff until my push2 stops working..

**Reply by zurie** [reply, marked as answer]
I removed OSCulator and now the push2 will send key commands without it.  Should be fine for Apple Silicon now.
